cinema/views.py

```diff
@@ -56,3 +92,48 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.action == "retrieve":
+            queryset = queryset.select_related("movie")
+
+        elif self.action == "list":
+            queryset = (
+                queryset
+                .select_related("cinema_hall")
+                .annotate(tickets_available=(
+                    F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                ))
+            ).order_by("id")
+
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        return queryset.distinct()
+
+
+class TicketViewSet(viewsets.ModelViewSet):
+    queryset = Ticket.objects.all()
+    serializer_class = TicketSerializer
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderPagination
+
+    def get_queryset(self):
+        return self.queryset.filter(user=self.request.user)
+
+    def perform_create(self, serializere):
```

Comment:
There is a typo in the `perform_create` method parameter name. It should be `serializer` instead of `serializere`. This typo will cause a `NameError` when trying to call `save` on `serializere`.